js/app/dashboard/redux/actions.js

```diff
@@ -1593,41 +1471,54 @@ export function createTour(tasks, name, date) {
   }
 }
 
-export function modifyTour(tour, tasks) {
+export function updateTourInUI(dispatch, tour, tasks) {
+  dispatch(disableDropInTours())
+  dispatch(modifyTourRequest(tour, tasks))
+}
 
-  return function(dispatch, getState) {
+export function modifyTour(tour, tasks, isTourUIAlreadyUpdated=false) {
+  /*
+    When assigning a task to a tour which is already is assigned,
+    `modifyTaskList` would be responsible for updating the UI.
+  */
 
-    dispatch(modifyTourRequest(tour, tasks))
+  return async function(dispatch, getState) {
+
+    if (!isTourUIAlreadyUpdated) updateTourInUI(dispatch, tour, tasks)
 
     const { jwt } = getState()
 
-    createClient(dispatch).request({
-      method: 'put',
-      url: tour['@id'],
-      data: {
-        name: tour.name,
-        tasks: _.map(tasks, t => t['@id'])
-      },
-      headers: {
-        'Authorization': `Bearer ${jwt}`,
-        'Accept': 'application/ld+json',
-        'Content-Type': 'application/ld+json'
-      }
-    })
-      .then(res => {
-        let _tour = res.data
+    try {
+      var response = await createClient(dispatch).request({
+        method: 'put',
+        url: tour['@id'],
+        data: {
+          name: tour.name,
+          tasks: _.map(tasks, t => t['@id'])
+        },
+        headers: {
+          'Authorization': `Bearer ${jwt}`,
+          'Accept': 'application/ld+json',
+          'Content-Type': 'application/ld+json'
+        }
+      })
+    } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error)
+        dispatch(modifyTourRequestError(tour))
+    }
+
+    return new Promise((resolve) => {
```

Comment:
Same as earlier. There is no need to create a Promise explicitly. As it's an async function JS will wrap returned result into a promise